dispatchers.go

```diff
@@ -354,10 +404,26 @@ func run(b *Builder, args []string, attributes map[string]bool, flagArgs []strin
 		}
 	}
 
+	var files []File
+	for _, heredoc := range heredocs {
+		command := parseCommandForHeredoc(original)
```

Comment:
Should we check for if command==""

Reply:
No its not needed, empty command is expected and code further honors it, there are test in `dispatcher_test.go` for both the cases and further test in buildah https://github.com/containers/buildah/pull/5092


Here empty command only means that treat default shell or `sh` as the default command.


For instance in command is parsed as empty which means implementation should use default shell and command.
```Dockerfile
RUN <<EOF
echo "Hello" >> /hello
echo "World!" >> /hello
EOF
```

Here command is parsed as `python3` instead of default shell
```Dockerfile
RUN python3 <<EOF
with open("/hello", "w") as f:
    print("Hello", file=f)
    print("Something", file=f)
EOF
```